tests/acceptance-tests/src/acceptance/http_forwarded_for_test.go

```diff
@@ -14,44 +14,50 @@ import (
 	"strings"
 )
 
-type Headers struct {
-	X_Forwarded_For   []string `json:"X-Forwarded-For"`
-	X_Forwarded_Proto []string `json:"X-Forwarded-Proto"`
-}
-
-var _ = PDescribe("X-Forwarded headers", func() {
-	var headers Headers
+var _ = Describe("X-Forwarded headers", func() {
+	var headers struct {
+		X_Forwarded_For []string `json:"X-Forwarded-For"`
+	}
 
 	var _ = BeforeEach(func() {
 		appName := generator.PrefixedRandomName("CATS-APP-")
 		Expect(cf.Cf(
 			"push", appName,
 			"-b", config.GoBuildpackName,
-			"-p", "../../example_apps/print_request_headers",
+			"-p", "../../apps/print_request_headers",
 			"-d", config.AppsDomain,
 			"-c", "./bin/debug_app; sleep 1; echo 'done'",
 		).Wait(CF_PUSH_TIMEOUT)).To(Exit(0))
 
-		jsonData := helpers.CurlAppRoot(appName)
-		headers = Headers{}
+		curlArgs := []string{"-f", "-H", "X-Forwarded-For: 1.2.3.4"}
+		jsonData := helpers.CurlApp(appName, "/", curlArgs...)
 
 		err := json.Unmarshal([]byte(jsonData), &headers)
 		Expect(err).NotTo(HaveOccurred())
 	})
 
 	It("should have an upstream IP that is not on our network", func() {
 		Expect(headers.X_Forwarded_For).Should(HaveLen(1))
+		remoteIP := strings.Split(headers.X_Forwarded_For[0], ",")[1]
+
+		parsedRemoteIP := net.ParseIP(strings.TrimSpace(remoteIP))
+
+		Expect(parsedRemoteIP).ToNot(BeNil())
 
-		remoteIP := strings.Split(headers.X_Forwarded_For[0], ",")[0]
-		parsedRemoteIP := net.ParseIP(remoteIP)
 		_, infra_cidr, _ := net.ParseCIDR("10.0.0.0/8")
 
 		Expect(infra_cidr.Contains(parsedRemoteIP)).To(BeFalse())
 	})
 
-	It("should indicate that the user connection was encrypted", func() {
-		Expect(headers.X_Forwarded_Proto).Should(HaveLen(1))
-		Expect(headers.X_Forwarded_Proto[0]).To(Equal("https"))
+	It("should append to existing X-Forwarded-For header", func() {
+		Expect(headers.X_Forwarded_For).Should(HaveLen(1))
+
+		remoteIP := strings.Split(headers.X_Forwarded_For[0], ",")[0]
+		parsedRemoteIP := net.ParseIP(remoteIP)
+
```

Comment:
Now, we always send X-Forwarded-For header via curl request. The detected IP is in different place as pre-set IP should be preserved. To be more specific: if you append to the X-Forwarded-For  header it will look like that: 
```
X-Forwarded-For: 1.2.3.4, 4.3.2.1, 6.6.6.6
```
where:
- 1.2.3.4 is preserved [0]
- 4.3.2.1 is detected by ELB [1]
- 6.6.6.6 is added by gorouter [2]

Note the spaces after commas - TrimSpace removes space from the string as when you split it by comma you are gonna get " 4.3.2.1" which is not parsable IP.